### PR TITLE
Fix a bug where calling a task with custom strategies would fail beca…

### PR DESCRIPTION
**Pull request by jeyrschabu**
…use of casting error

- Added a guard to ensure the parent task is a pipeline before attempting to cast to a Pipeline

### CODE REVIEW

**Comment by jeyrschabu** [comment]
@cfieber @ajordens PTAL

File: orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/CreateServerGroupTaskSpec.groovy [hunk 1]
```diff
@@ -214,6 +215,7 @@ class CreateServerGroupTaskSpec extends Specification {
 
     def childTrigger = [
       parentExecution: [
+        isPipeline: true,
```
Comment (tomaslin) [inline orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/CreateServerGroupTaskSpec.groovy]: is this actually failing a test? Think only the parentTrigger needs isPipeline

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/model/Pipeline.groovy [hunk 2]
```diff
@@ -16,9 +16,11 @@
 
 package com.netflix.spinnaker.orca.pipeline.model
 
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
 import groovy.transform.CompileStatic
 
 @CompileStatic
+@JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (cfieber) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/model/Pipeline.groovy]: what properties were landing in as a Pipeline that we need to ignore? (concerned we are deserializing an Orchestration somewhere to a Pipeline and not to a generic Execution and wondering what the ripple effect of that is)

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy [hunk 3]
```diff
@@ -133,7 +133,7 @@ trait DeploymentDetailsAware {
     if (execution instanceof Pipeline) {
       if (execution.trigger.parentExecution instanceof Pipeline) {
         return execution.trigger.parentExecution
-      } else if (execution.trigger.parentExecution) {
+      } else if (execution.trigger.parentExecution?.isPipeline) {
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy]: Pull this up to `execution.trigger` and take out the `JsonIgnoreProperties`, and I think we'd be good.

Should be easy enough to test this.

**Comment by jeyrschabu** [comment]
@cfieber @ajordens PTAL.

**Comment by ajordens** [comment]
LGTM. I'm not sure what the release process is for this branch but I'm sure @cfieber can fill us in when he has a moment.

**Comment by jeyrschabu** [comment]
@cfieber can confirm the steps he provided earlier:

To Release:
Select release/1.237.x as the release branch and set it to v1.237.1

Going to merge this in the meantime

**Pull request merged**